src/Playwright/Core/Locator.cs

```diff
@@ -511,6 +517,9 @@ private static string EscapeForRegex(string text)
         var patern = new Regex(@"[.*+?^>${}()|[\]\\]");
         return patern.Replace(text, "\\$&");
     }
+
+    async Task<IReadOnlyList<ILocator>> ILocator.AllAsync()
+        => new string[await CountAsync()].Select((e, i) => Nth(i)).ToArray();
```

Comment:
This might be more readable (at least it is for me):
```suggestion
        => Enumerable.Range(0, await CountAsync()).Select(Nth).ToArray();
```